scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/package.scala

```diff
@@ -0,0 +1,88 @@
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.scio.parquet
+
+import java.lang.{Boolean => JBoolean}
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.util.ScioUtil
+import com.spotify.scio.values.SCollection
+import org.apache.avro.Schema
+import org.apache.avro.specific.SpecificRecordBase
+import org.apache.beam.sdk.io.hadoop.inputformat.HadoopInputFormatIO
+import org.apache.beam.sdk.transforms.SimpleFunction
+import org.apache.hadoop.mapreduce.Job
+import org.apache.hadoop.mapreduce.lib.input.FileInputFormat
+import org.apache.parquet.avro.AvroParquetInputFormat
+import org.apache.parquet.filter2.predicate.FilterPredicate
+import org.apache.parquet.hadoop.ParquetInputFormat
+
+import scala.reflect.ClassTag
+
+package object avro {
+
+  implicit class ParquetAvroScioContext(val self: ScioContext) extends AnyVal {
+    def parquetAvroFile[T <: SpecificRecordBase : ClassTag](path: String,
+                                                            projection: Schema = null,
+                                                            predicate: FilterPredicate = null)
+    : SCollection[T] = {
+      val cls = ScioUtil.classOf[T]
+      val readSchema = cls.getMethod("getClassSchema").invoke(null).asInstanceOf[Schema]
+
+      val job = Job.getInstance()
+
+      // Client id/secret of Google-managed project associated with the Cloud SDK
+      // These are needed since `FileInputFormat.setInputPaths` validates paths locally and
+      // requires the user's GCP credentials.
+      job.getConfiguration.setBoolean("fs.gs.auth.service.account.enable", false)
+      job.getConfiguration.set("fs.gs.auth.client.id", "32555940559.apps.googleusercontent.com")
+      job.getConfiguration.set("fs.gs.auth.client.secret", "ZmssLNjJy2998hD4CTg2ejr2")
+
+      FileInputFormat.setInputPaths(job, path)
+
+      // These will interfere with credentials in Dataflow workers
+      if (!ScioUtil.isLocalRunner(self.options.getRunner)) {
+        job.getConfiguration.unset("fs.gs.auth.service.account.enable")
+        job.getConfiguration.unset("fs.gs.auth.client.id")
+        job.getConfiguration.unset("fs.gs.auth.client.secret")
+      }
+
+      job.setInputFormatClass(classOf[AvroParquetInputFormat[T]])
+      job.getConfiguration.setClass("key.class", classOf[Void], classOf[Void])
+      job.getConfiguration.setClass("value.class", cls, cls)
+
+      AvroParquetInputFormat.setAvroReadSchema(job, readSchema)
+      if (projection != null) {
+        AvroParquetInputFormat.setRequestedProjection(job, projection)
+      }
+      if (predicate != null) {
+        ParquetInputFormat.setFilterPredicate(job.getConfiguration, predicate)
+      }
+
+      val source = HadoopInputFormatIO.read[JBoolean, T]()
```

Comment:
`null` causes NPE in https://github.com/apache/beam/blob/master/sdks/java/io/hadoop/input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java#L714